### DISCUSSION TITLE
Is it possible to include a client certificate in my requests?

### DISCUSSION THREAD

**Discussion by uchoa** [rest-nvim/rest.nvim#467] [Q&A]
I need to test some endpoints that require a client certificate to be included for authentication. How do I include it in rest.nvim?  Since it is possible with curl, I suppose it can be done in rest.nvim as well.

**Comment by uchoa** [comment]
The command line version of it would be:
`curl --cert certificate-file.pem --key private-key.pem https://someurl.com`

**Comment by boltlessengineer** [comment]
This is now possible with #519 merged.

```lua
vim.g.rest_nvim = {
  clients = {
    curl = {
      opts = {
        certificates = {
          ["someurl.com"] = {
            set_certificate_crt = "/path/to/certiciate-file.pem",
            set_certificate_key = "/path/to/private-key.pme",
          },
        },
      },
    },
  },
}
```

See `:h rest.Opts.Clients.Curl.Opts` for more information.
Also you can find some examples from #528
